crates/chia-bls/src/cached_bls.rs

```diff
@@ -78,6 +78,40 @@ impl BlsCache {
 
         aggregate_verify_gt(sig, iter)
     }
+
+    pub fn aggregate_verify_with_iter(
```

Comment:
rather than duplicating `aggregate_verify()`, I think you could just change it to take an iterator of pairs, instead of two iterators. basically, replace it with this function. Tests and the python bindings would have to be updated, but you'd need tests for this function anyway

Reply:
Done